aws/resource_aws_workspaces_directory_test.go

```diff
@@ -2,16 +2,87 @@ package aws
 
 import (
 	"fmt"
-	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"log"
 	"reflect"
 	"testing"
+	"time"
 
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/workspaces"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
 	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
 	"github.com/hashicorp/terraform-plugin-sdk/terraform"
 )
 
+func init() {
+	resource.AddTestSweepers("aws_workspaces_directory", &resource.Sweeper{
+		Name: "aws_workspaces_directory",
+		F:    testSweepWorkspacesDirectories,
+	})
+}
+
+func testSweepWorkspacesDirectories(region string) error {
+	client, err := sharedClientForRegion(region)
+	if err != nil {
+		return fmt.Errorf("error getting client: %s", err)
+	}
+	conn := client.(*AWSClient).workspacesconn
+
+	input := &workspaces.DescribeWorkspaceDirectoriesInput{}
+	for {
+		resp, err := conn.DescribeWorkspaceDirectories(input)
```

Comment:
Nit: The AWS Go SDK provides a paginated function for this call: [`DescribeWorkspaceDirectoriesPages`](https://docs.aws.amazon.com/sdk-for-go/api/service/workspaces/#WorkSpaces.DescribeWorkspaceDirectoriesPages)